lib/Basic/UUID.cpp

```diff
@@ -15,40 +15,126 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include <uuid/uuid.h>
 #include "swift/Basic/UUID.h"
 
+// WIN32 doesn't natively support <uuid/uuid.h>. Instead, we use Win32 APIs
+// class
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#include <rpc.h>
+#include <string>
+
+// Windows has a different layout of UUID than libuuid. However, they are the same
+// size, as mandated by the UUID specification (RFC 4122). This means that we can
+// convert the memory of a 128 bit unsigned char array to a ::UUID.
+// We also can't expose <rpc.h> in any headers, as it imports <windows.h> which
+// defines macros that conflict with many definitions (e.g. ERROR, std::max)
+union Conversion {
```

Comment:
I'm not sure I can entirely remove this. We use this conversion union in methods such as `UUIDCompare` and `UUIDToString` that require a UUID passed. I have removed all the conversions from memcpy though as you suggested.